Library/Homebrew/os.rb

```diff
@@ -1,6 +1,6 @@
 module OS
   def self.mac?
-    /darwin/i === RUBY_PLATFORM
+    /darwin/i === RUBY_PLATFORM || ENV["HOMEBREW_EMULATE_MAC"]
```

Comment:
We don't really need this. It should works fine without it. 

In fact, as a design guidance, we should allow homebrew to run under Linux for metadata processing(e.g. `brew info --json=v1`) without any environment.